### PR TITLE
Adding a marginalkde plot much like marginalhist

### PR DESCRIPTION
**Pull request by farr**
This PR adds a `marginalkde` plot much like the `marginalhist` plot.

### CODE REVIEW

**Comment by BeastyBlacksmith** [comment]
Can you show an example?

**Comment by farr** [comment]
```julia
marginalkde(randn(1024), randn(1024))
```

marginalkde

**Comment by farr** [comment]
Or, maybe more illustrative:

```julia
x = randn(1024)
y = randn(1024)
marginalkde(x, x+y)
```

correlated-marg

File: src/marginalkde.jl [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    x,y = kc.args
+
+    x = vec(x)
+    y = vec(y)
+
+    m_x = median(x)
+    m_y = median(y)
+
+    dx_l = m_x - quantile(x, 0.16)
+    dx_h = quantile(x, 0.84) - m_x
```
Comment (daschw) [inline src/marginalkde.jl:13]: Are 0.16 and 0.84 the best choice in all cases, or should we add an additional kwarg or alternatively specify `xlims` and `ylims` using `-->` in the recipe?

Comment (farr) [reply src/marginalkde.jl:13]: Hmm.  That's controllable via the `clip` argument (which multiplies these values); those numbers would represent +/- 1-sigma in a Gaussian, so are probably sensible for choosing a "scale" that gets multiplied by the `clip` argument.

Comment (daschw) [reply src/marginalkde.jl:13]: Oh, sorry, I did not take a close enough look.

**Review by daschw — APPROVED** [review]
Looks great!

**Comment by farr** [comment]
Great---thanks!

**Pull request merged**